common/src/gas_provider/error.rs

```diff
@@ -0,0 +1,53 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+/// Trait represents possible errors of Tree.
+///
+/// Contains constructors for all existing errors.
+pub trait Error {
+    /// Gas (gas tree) has already been created for the provided key.
+    fn node_already_exists() -> Self;
+
+    /// Errors stating that gas tree has been invalidated
+
+    /// Parent must be in the tree, but not found
+    ///
+    /// This differs from `node_not_found`, because parent
+    /// node for local node types must be found, but was not. Thus,
+    /// tree is invalidated.
+    fn parent_is_lost() -> Self;
+
+    /// Parent node must have children, but they weren't found
+    ///
+    /// If node is a parent to some other node it must have at least
+    /// one child, otherwise it's id can't be used as a parent for
+    /// local nodes in the tree.
+    fn parent_has_no_children() -> Self;
+
+    /// Value node doesn't exist for a key
+    fn node_not_found() -> Self;
+
+    /// Procedure can't be called on consumed node
```

Comment:
```suggestion
    /// Procedure can't be called on consumed node.
```

Dots at the end of each doc comment.